### ISSUE TITLE
Crestron NVX Routing (Via DM NVX Director)

### ISSUE THREAD

**Issue by CountParadox** [bitfocus/companion-module-requests#1135]
- [x] **I have researched the list of existing Companion modules and requests and have determined this has not yet been requested**

_The name of the device, hardware, or software you would like to control:_

**Crestron NVX Director managed NVX networks (HDMI Matrices over IP)**

_What you would like to be able to make it do from Companion:_

**Route a specified input to one or more specified outputs.**

_Direct links or attachments to the ethernet control protocol or API:_

There appears to be both a REST API and a Websockets API:
https://sdkcon78221.crestron.com/sdk/DM_NVX_REST_API/Content/Topics/Home.htm

Some notes: 
https://www.reddit.com/r/crestron/comments/nxlx4i/nvx_rest_api/

Happy to **~~Offer a reward of AUD75~~**

**Comment by josephdadams** [comment]
@jtf4

**Comment by JTF4** [comment]
I'd love to discuss doing this module on commission, but $50 usd would not be enough. 

If you want to discuss further, feel free to PM me on the Slack (@David Stevens) or opensource@jtf4.com

**Comment by jcracken** [comment]
Was any progress ever made on this request? Have interest in getting this over the line as well.

**Comment by CountParadox** [comment]
No, my programmer added logic into the crestron processor that accepted a simpler API I could bash out myself;

Something akin to http calls to 

 x.x.x.x/nvx/12/17

To route input 12 to 17

**Comment by josephdadams** [comment]
This is now available as the **`crestron-nvx`** module (latest version v1.0.1).

You can install it from Companion's module store, or see https://bitfocus.io/connections/crestron-nvx

Closing as implemented — please open a new issue on [the module's repository](https://github.com/bitfocus/companion-module-crestron-nvx/issues) if something specific is missing.
